### PR TITLE
Changes in obstacle avoidance module

### PR DESCRIPTION
**Pull request by ClintNous**
@rmeertens Zou jij er nog even naar kunnen kijken?

### CODE REVIEW

**Comment by gautierhattenberger** [comment]
Would be nice to provide some descriptions of the changes when making a pull request. Otherwise, @fvantienen @OpenUAS if you know this work, I let you merge it.

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c [hunk 1]
```diff
@@ -52,175 +53,146 @@
 #ifndef AVOIDANCE_WIDTH_IN_MEASUREMENT_VALUES
 #define AVOIDANCE_WIDTH_IN_MEASUREMENT_VALUES 6
 #endif
```
Comment (rmeertens) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: What does this mean?

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c [hunk 2]
```diff
@@ -52,175 +53,146 @@
...
+//#ifndef PRINT_MATRIX
+//#define PRINT_MATRIX
+//#endif
+
+//#ifndef SEND_OA_DATA
+//#define SEND_OA_DATA
+//#endif
+
+//Initialize variables which can be changed in settings
+//OA_general
```
Comment (rmeertens) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: Can we remove these?

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c [hunk 3]
```diff
@@ -52,175 +53,146 @@
...
+//Vector_method
 float F1 = 0.1;
 float F2 = 0.9;
 float Cfreq = 1;
-float Ko = 60;//1000;//10000
-float Kg = 100;//100;//100
-float Dist_offset = 0;//-0.125;
+float Ko = 60;
+float Kg = 100;
+float Dist_offset = 0;
```
Comment (rmeertens) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: More descriptive variable names??

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c [hunk 4]
```diff
@@ -52,175 +53,146 @@
...
-float  K_goal = 1;
+//Pot_method
 float  K_obst = 40;
+float  K_goal = 1;
+float  b_damp = 0;
 float  c1_oa = 0.4;
 float  c2_oa = 0.4;
 float  c3_oa = 5.0;
 float  c4_oa = 0.5;
 float  c5_oa = 0.9;
```
Comment (rmeertens) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: More descriptive variable names (put them together in a struct would also be fine.

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c [hunk 5]
```diff
@@ -368,29 +371,29 @@ void pingpong_euler(float *distances_hor_local, float *horizontalAnglesMeasureme
...
   } else {
     ref_pitch = sumPitch;
   }
 
 
-  if (sumRoll > attitude_reference_roll) {
-    ref_roll = attitude_reference_roll;
-  } else if (sumRoll < -attitude_reference_roll) {
-    ref_roll = -attitude_reference_roll;
+  if (sumRoll > reference_roll) {
```
Comment (rmeertens) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: apparently sumRoll should be sum_roll (something with codestyle)

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c [hunk 6]
```diff
@@ -864,65 +773,37 @@ void CN_vector_escape_velocity(void)
...
   else if (V_total > v_max && obstacle_flag == 0) {
     escape_flag = 0;
   }
-
-  printf("V_total: %f", V_total);
-  printf("escape_flag: %i, obstacle_flag: %i, set_bias: %i", escape_flag, obstacle_flag, set_bias);
-  printf("escape_flag: %i, obstacle_flag: %i, set_bias: %i", escape_flag, obstacle_flag, set_bias);
+  //printf("V_total: %f", V_total);
+  //printf("escape_flag: %i, obstacle_flag: %i, set_bias: %i", escape_flag, obstacle_flag, set_bias);
+  //printf("escape_flag: %i, obstacle_flag: %i, set_bias: %i", escape_flag, obstacle_flag, set_bias);
```
Comment (rmeertens) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: Can we remove this?

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.h [hunk 7]
```diff
@@ -70,4 +29,21 @@ extern float c5_oa;
 extern float kv;
 extern float epsilon;
 
+extern void serial_init(void);
+extern void serial_update(void);
+extern void serial_start(void);
+void setAnglesMeasurements(float *anglesMeasurements, float *anglesMeasurements_ver, float *centersensorRad, float *fieldOfViewRad,
+                           uint16_t *size_matrix_local);
+extern void matrix_2_pingpong(float *distancesMeters, uint16_t *size_matrix, float *distances_hor);
+extern void kalman_filter_distances(float *distancesMeters);
```
Comment (rmeertens) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.h]: How does this function know how many distances there are in the array? Can we add an integer that says how long this array is?

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c (hunk 1 above)
Comment (ClintNous) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: Amount of horizontal measurements for a detection sensor

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.h (hunk 7 above)
Comment (ClintNous) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.h]: It uses stereocam_data.len

File: sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c (hunk 4 above)
Comment (ClintNous) [inline sw/airborne/modules/obstacle_avoidance/obstacle_avoidance.c]: Many variables would have to be changed, I added descriptions in the header file instead. Hopefully that is sufficient

**Comment by rmeertens** [comment]
Hey guys, 

As this student is gone now, I think we should accept these latest improvements!

**Comment by flixr** [comment]
@rmeertens could you clean this up and test it and make a new pull request then?

**Comment by flixr** [comment]
@rmeertens unless you (or someone else) will clean this up any time soon, I'll close this...

**Comment by rmeertens** [comment]
Too bad, I don't think anybody is going to clean it up :( 
Maybe close it (Or ask @ClintNous for a last cleanup)

**Comment by rmeertens** [comment]
As for mergability -> the only conflict is conf/conf_example.xml .... all other changes are improvements bringing the code quality a bit higher.

**Comment by flixr** [comment]
Not quite... this also adds things like trying to send OA_DATA, which is a message we don't have and removes copy right headers... also indentation needs to be cleaned up...

**Comment by podhrmic** [comment]
@flixr @rmeertens any updates here? Definitely is incomplete - do you guys still want to work on it, or should we close it?

**Comment by EwoudSmeur** [comment]
Both @ClintNous and @rmeertens have left the MAVLab, so I am guessing they will not continue this. I am fine with closing.

**Comment by rmeertens** [comment]
Yes, too bad. Also, I don't think anyone in the MAVLab is doing obstacle avoidance nowadays, so there is no good person to take over this pull request. I still love to help out, but don't think fixing this is a high priority for me :( let's close it,

**Comment by ClintNous** [comment]
Hi All,

Sorry for late reply, currently I also don't have the time to look into it.
So good that it is closed now.

@Ewoud, Let me know if a student start doing OA stuff again, than maybe I
can help him with some code.

Greets ,

Clint

**Pull request closed without merging**